langchain4j-core/src/main/java/dev/langchain4j/data/document/Metadata.java

```diff
@@ -124,6 +139,30 @@ public String getString(String key) {
             "It cannot be returned as a String.", key, value, value.getClass().getName());
     }
 
+    /**
+     * Returns the collection of {@code String} values associated with the given key.
+     *
+     * @param key the key
+     * @return the collection of {@code String} values associated with the given key, or an empty collection if the key is not present.
+     * @throws RuntimeException if the value is not of type Collection
+     */
+    public Collection<String> getStrings(String key) {
+        if (!containsKey(key)) {
+            return Collections.emptyList();
+        }
+
+        Object values = metadata.get(key);
```

Comment:
The double lookup in the map could be avoided (and yes this should be applied to all other getters in this class).

```suggestion
        Object values = metadata.get(key);
        if (values == null) {
            return Collections.emptyList();
        }
```